ext/json/ext/parser/parser.c

```diff
@@ -2363,9 +2363,17 @@ static VALUE json_string_unescape(char *string, char *stringEnd, int intern, int
 	char buf[4];
 
 	if (bufferSize > MAX_STACK_BUFFER_SIZE) {
+# ifdef HAVE_RB_ENC_INTERNED_STR
```

Comment:
note, i've wrapped the fix with this `HAVE_RB_ENC_INTERNED_STR` guard as the issue only pertains to the call to `rb_enc_interned_str` below. Saves allocating the extra byte unnecessarily. Although, if there was concern about the extra allocation this would likely look like:
```
bufferStart = buffer = ALLOC_N(char, !intern || bufferSize ? bufferSize : 1);
```

Or would you prefer for simplicities sake to just go with the suggestion in #475 from @jeremyevans  (ie without the `HAVE_RB_ENC_INTERNED_STR` guards)?

Of course the safest solution would be to simply allocate room for the null terminator all of the time (`bufferSize + 1`) but that's likely unnecessary given how the pointers are being used.